packages/backend/src/ee/services/ai/agents/agent.ts

```diff
@@ -88,44 +90,54 @@ export const runAgent = async ({
 
     const model = getOpenaiGpt41model(args.openaiApiKey);
 
-    const result = await generateText({
-        model,
-        tools,
-        toolChoice: 'auto',
-        messages,
-        maxSteps: 10,
-        maxRetries: 3,
-        temperature: 0.2,
-        experimental_repairToolCall: async ({
-            messages: conversationHistory,
-            error,
-            toolCall,
-            parameterSchema,
-        }) => {
-            if (NoSuchToolError.isInstance(error)) {
-                return null;
-            }
-
-            const tool = tools[toolCall.toolName as keyof typeof tools];
-
-            // TODO: extract this as separate agent
-            const { object: repairedArgs } = await generateObject({
-                model,
-                schema: tool.parameters as ZodType<AnyType>,
+    try {
+        const result = await generateText({
+            model,
+            tools,
+            toolChoice: 'auto',
+            messages,
+            maxSteps: 10,
+            maxRetries: 3,
+            temperature: 0.2,
+            experimental_repairToolCall: async ({
                 messages: conversationHistory,
-                prompt: [
-                    `The model tried to call the tool "${toolCall.toolName}"` +
-                        ` with the following arguments:`,
-                    JSON.stringify(toolCall.args),
-                    `The tool accepts the following schema:`,
-                    JSON.stringify(parameterSchema(toolCall)),
-                    'Please fix the arguments.',
-                ].join('\n'),
-            });
-
-            return { ...toolCall, args: JSON.stringify(repairedArgs) };
-        },
-    });
-
-    return result.text;
+                error,
+                toolCall,
+                parameterSchema,
+            }) => {
+                if (NoSuchToolError.isInstance(error)) {
+                    return null;
+                }
+
+                const tool = tools[toolCall.toolName as keyof typeof tools];
+
+                // TODO: extract this as separate agent
+                const { object: repairedArgs } = await generateObject({
+                    model,
+                    schema: tool.parameters as ZodType<AnyType>,
+                    messages: [
+                        ...conversationHistory,
+                        {
+                            role: 'system',
+                            content: [
+                                `The model tried to call the tool "${toolCall.toolName}"` +
+                                    ` with the following arguments:`,
+                                JSON.stringify(toolCall.args),
+                                `The tool accepts the following schema:`,
+                                JSON.stringify(parameterSchema(toolCall)),
+                                'Please fix the arguments.',
+                            ].join('\n'),
+                        },
+                    ],
```

Comment:
both `messages` and `prompt` cannot be used, it was throwing an error

`Error repairing tool call: Invalid prompt: prompt and messages cannot be defined at the same time`